File: solutions/python/03/ncr/main.py
MOD = 998244353


def build_ncr(n: int, mod: int = MOD) -> list:
    c = [[0] * (n + 1) for _ in range(n + 1)]
    for i in range(n + 1):
        c[i][0] = c[i][i] = 1
        for j in range(1, i):
            c[i][j] = (c[i - 1][j - 1] + c[i - 1][j]) % mod
    return c


def main() -> None:
    c = build_ncr(10)
    print(c[5][2])


if __name__ == "__main__":
    main()

File: solutions/go/03/ncr/main.go
package main

import "fmt"

const mod = 998244353

func main() {
	n := 10
	c := make([][]int, n+1)
	for i := 0; i <= n; i++ {
		c[i] = make([]int, n+1)
		c[i][0], c[i][i] = 1, 1
		for j := 1; j < i; j++ {
			c[i][j] = (c[i-1][j-1] + c[i-1][j]) % mod
		}
	}
	fmt.Println(c[5][2])
}
